source/utils/timed-out.ts

```diff
@@ -91,7 +94,7 @@ export default (request: ClientRequest, delays: Delays, options: TimedOutOptions
 
 	if (delays.socket !== undefined) {
 		const socketTimeoutHandler = (): void => {
-			timeoutHandler(delays.socket, 'socket');
+			timeoutHandler(delays.socket!, 'socket');
```

Comment:
Shouldn't if if statement act as a type guard, thus removing the need for the non-null assertion?

Reply:
Already fixed in master: https://github.com/sindresorhus/got/commit/640c30bbe37fa399a6f9c96357d897f45e54f52f#diff-7ab9a52ac2fccf434c1a34ce48f2883bL95-R95

Reply:
Ah, my mistake! :+1: